buf/registry/module/v1/file_descriptor_set_service.proto

```diff
@@ -0,0 +1,104 @@
+  //     For example, if the ResourceRef specifies "buf.build/foo/bar", which depends on
+  //     "buf.build/foo/baz", all files from "buf.build/foo/baz" will be imports.
+  //   - It is a Well-Known Type, and the Well-Known Type was not part of the Module specified
+  //     via the ResourceRef used in the request. Well-Known Types are special in the Protobuf ecosystem,
+  //     and can be imported without being contained in a specified module.
+  //   - It contains types that are referenced by types specified in the request. For example,
+  //     if the request specified symbol "foo.v1.Foo" contained within "foo.proto", which has a message
+  //     field of type "baz.v1.Baz", which is contained within "baz.proto", then the file "baz.proto"
+  //     will be marked as an import, regardless of which Module it derived from. If the types field
+  //     is not specified on the request, this situation will never occur.
+  //
+  // Note that if imports are excluded, the returned FileDescriptorSet may not (and likely will not)
+  // be self-contained. If imports are included, all types referenced within the FileDescriptorSet
+  // are contained within the returned FileDescriptorSet.
+  bool exclude_imports = 2;
+  // Include SourceCodeInfo in the returned FileDescriptorProtos.
+  //
+  // SourceCodeInfo is optional additional information that is not required to describe a set
+  // of Protobuf APIs.
+  bool include_source_code_info = 3;
+  // Include source-retention options in the returned FileDescriptorProtos.
+  //
+  // With the new Editions work, options are either runtime-retention or source-retention.
+  // Setting this option will include the latter.
+  bool include_source_retention_options = 4;
+  // Only include specific types (and types that they reference, if exclude_imports is not set).
+  //
+  // A symbol is a package, message, enum, service, method, or extension.
+  //
+  // Types are referenced by their fully-qualified name. For example, if "Foo" is a message within
+  // the package "foo.v1", the fully-qualified name is "foo.v1.Foo". Fully-qualfied names should
+  // not start with a period, that is specify "foo.v1.Foo", not ".foo.v1.Foo".
+  //
+  // If no types are provided, all types are returned.
+  repeated string types = 5;
+  // An explicit, opt-in field mask for the fields populated on the returned FileDescriptorProtos.
+  //
+  // For example, if you would only like to return the names of messages, use the mask:
+  //
+  //   {
+  //     paths: [
+  //       "message_type.name",
+  //       "message_type.nested_type.name"
+  //     ]
+  //   }
+  //
+  // The filters exclude_imports, exclude_source_code_info, exclude_source_retention_options and
+  // types continue to be respected even if this field is set. Specifically, if exclude_source_code_info
+  // is set, but this mask includes source_code_info, SourceCodeInfo will continue to be excluded.
+  google.protobuf.FieldMask field_mask = 6;
```

Comment:
Are we adding this to a lot of RPCs? Seems like we could leave out, especially since it will be error prone. For example, the sample given for message names only can return top-level names, not names for nested types. The thing Alfred is working on in bufstream will be a lot more capable. Also, it's feels awkward to have to add this to every request when we could instead implement this in a cross-cutting way in the BSR by accepting the mask in a header.

Reply:
Can remove for v1 of this